packages/interactivity/src/utils.ts

```diff
@@ -203,21 +228,31 @@ export function useCallback( callback, inputs ) {
  * available so functions like `getElement()` and `getContext()` can be used
  * inside the passed factory function.
  *
- * @param {Function} factory Imperative function that can return a cleanup
- *                           function.
- * @param {any[]}    inputs  If present, effect will only activate if the
- *                           values in the list change (using `===`).
+ * @param factory Imperative function that can return a cleanup
+ *                function.
+ * @param inputs  If present, effect will only activate if the
+ *                values in the list change (using `===`).
  */
-export function useMemo( factory, inputs ) {
+export function useMemo( factory: Function, inputs: any[] ) {
 	_useMemo( withScope( factory ), inputs );
 }
 
-// For wrapperless hydration.
-// See https://gist.github.com/developit/f4c67a2ede71dc2fab7f357f39cff28c
-export const createRootFragment = ( parent, replaceNode ) => {
+/**
+ * Creates a root fragment by replacing a node or an array of nodes in a parent element.
+ * For wrapperless hydration.
+ * See https://gist.github.com/developit/f4c67a2ede71dc2fab7f357f39cff28c
+ *
+ * @param parent      - The parent element where the nodes will be replaced.
+ * @param replaceNode - The node or array of nodes to replace in the parent element.
+ * @return The created root fragment.
+ */
+export const createRootFragment = (
+	parent: ParentElement,
+	replaceNode: Node | Node[]
+) => {
 	replaceNode = [].concat( replaceNode );
 	const s = replaceNode[ replaceNode.length - 1 ].nextSibling;
-	function insert( c, r ) {
+	function insert( c: any, r: any ) {
```

Comment:
Optional task: it would be helpful to unfold what `c`, `r` and `s` mean.

Reply:
cc @DAreRodz

Reply:
It's all from https://gist.github.com/developit/f4c67a2ede71dc2fab7f357f39cff28c, I think @c4rl0sbr4v0 is probably right.

Reply:
There is a TypeScripted version in case you need:

https://gist.github.com/developit/f4c67a2ede71dc2fab7f357f39cff28c?permalink_comment_id=4558744#gistcomment-4558744